src/components/PivotTable/PivotTableRenderer/util.ts

```diff
@@ -0,0 +1,157 @@
+import { MutableRefObject, SetStateAction } from 'react'
+import { KeyMap } from '../model/model-keyMap'
+import {
+  Dictionary,
+  IGNORED_TREE_KEYS,
+  InitialPosition,
+  NestingResult,
+  SpanValue,
+  StackObj,
+  TreeRoot,
+} from './model-pivotTableRenderer'
+import { GroupResult } from './classes/GroupResult'
+
+export const RESULT_PATH_KEY = 'RESULT'
+export const TOTAL = 'Total'
+const PATH_SEPARATOR = '|'
+
+export function getCurrentPath(filterKey: string, value: string): string {
+  return PATH_SEPARATOR + filterKey + '.' + value
+}
+
+export function getInitialPosition(ini: InitialPosition | undefined) {
+  const stack: InitialPosition[] = []
+
+  if (ini) {
+    stack.push(ini)
+  } else {
+    return 0
+  }
+
+  let result = 1
+
+  while (stack.length) {
+    const i = stack.pop()
+    if (i) {
+      if (i.spanAux) {
+        result += i.spanAux.value
+      }
+      if (i.auxIni) {
+        stack.push(i.auxIni)
+      } else if (i.parentIni) {
+        stack.push(i.parentIni)
+      }
+    }
+  }
+
+  return result
+}
+
+export function getResult<T extends object>(
+  data: Dictionary<T> & TreeRoot,
+  increment: 'column' | 'row',
+  keyMapping: KeyMap<T>,
+  onlyIncreaseSpanOnKeys?: Array<keyof T>
+): NestingResult<T>[] {
+  const stack: StackObj[] = []
+
+  stack.push({ data: data, [increment]: 1 })
+
+  const result: NestingResult<T>[] = []
+
+  while (stack.length) {
+    const obj = stack.shift()
+    if (obj) {
+      let spanAux: SpanValue
+      let iniAux: InitialPosition
+      const currentNodeKey = obj.data.nodeKey
+      const increaseSpan = !onlyIncreaseSpanOnKeys || onlyIncreaseSpanOnKeys.includes(currentNodeKey)
+      Object.keys(obj.data)
+        .filter((k) => !IGNORED_TREE_KEYS.includes(k))
+        .sort(keyMapping.get(currentNodeKey)?.ordenator)
+        .forEach((key) => {
+          let span = { value: 1 }
+          let spanTree = [span]
+          if (obj.spanTree) {
+            const myKeys = Object.keys(obj.data).filter((k) => !IGNORED_TREE_KEYS.includes(k))
+            const lastSpan = obj.spanTree[0]
+            if (myKeys.length > lastSpan.value && increaseSpan) {
+              for (let i = 0; i < obj.spanTree.length; i++) {
+                obj.spanTree[i].value++
+              }
+            }
+            spanTree.push(...obj.spanTree)
+          }
```

Comment:
I don't think this is a good method to separate, only because line 80 has side effects on the `initialPosition` on line 95 and 127. It would be harder to understand if this code was in another method.